src/transformers/models/clip/modeling_clip.py

```diff
@@ -0,0 +1,938 @@
+    """
+    Args:
+        loss (:obj:`torch.FloatTensor` of shape :obj:`(1,)`, `optional`, returned when :obj:`return_loss` is :obj:`True`):
+            Contrastive loss for image-text similarity.
+        logits_per_image:(:obj:`torch.FloatTensor` of shape :obj:`(image_batch_size, text_batch_size)`):
+            The scaled dot product scores between :obj:`image_embeds` and :obj:`text_embeds`. This represents the
+            image-text similarity scores.
+        logits_per_text:(:obj:`torch.FloatTensor` of shape :obj:`(text_batch_size, image_batch_size)`):
+            The scaled dot product scores between :obj:`text_embeds` and :obj:`image_embeds`. This represents the
+            text-image similarity scores.
+        text_embeds(:obj:`torch.FloatTensor` of shape :obj:`(batch_size, output_dim`):
+            The text embeddings obtained by applying the projection layer to the pooled output of
+            :class:`~transformers.CLIPTextModel`.
+        image_embeds(:obj:`torch.FloatTensor` of shape :obj:`(batch_size, output_dim`):
+            The image embeddings obtained by applying the projection layer to the pooled output of
+            :class:`~transformers.CLIPVisionModel`.
+        text_model_output(:obj:`BaseModelOutputWithPooling`):
+            The output of the :class:`~transformers.CLIPTextModel`.
+        vision_model_output(:obj:`BaseModelOutputWithPooling`):
+            The output of the :class:`~transformers.CLIPVisionModel`.
+    """
+
+    loss: Optional[torch.FloatTensor] = None
+    logits_per_image: torch.FloatTensor = None
+    logits_per_text: torch.FloatTensor = None
+    text_embeds: torch.FloatTensor = None
+    image_embeds: torch.FloatTensor = None
+    text_model_output: BaseModelOutputWithPooling = None
+    vision_model_output: BaseModelOutputWithPooling = None
+
+
+class QuickGELU(nn.Module):
+    def forward(self, hidden_states: torch.Tensor):
+        return hidden_states * torch.sigmoid(1.702 * hidden_states)
+
+
+class CLIPVisionEmbeddings(nn.Module):
+    def __init__(self, config: CLIPVisionConfig):
+        super().__init__()
+        self.config = config
+        self.embed_dim = config.hidden_size
+        self.image_size = config.image_size
+        self.patch_size = config.patch_size
+
+        scale = self.embed_dim ** -0.5
+        class_embed_weights = torch.empty(self.embed_dim)
+        nn.init.normal_(class_embed_weights, mean=0.0, std=scale)
+        self.class_embedding = nn.Parameter(class_embed_weights)
+
+        self.patch_embedding = nn.Conv2d(
+            in_channels=3, out_channels=self.embed_dim, kernel_size=self.patch_size, stride=self.patch_size, bias=False
+        )
+
+        self.num_patches = (self.image_size // self.patch_size) ** 2
+        self.num_positions = self.num_patches + 1
+        self.position_embedding = nn.Embedding(self.num_positions, self.embed_dim)
+        self.register_buffer("position_ids", torch.arange(self.num_positions).expand((1, -1)))
```

Comment:
not sure - isn't `torch.device` needed in the `torch.arange(...)` ?

Reply:
No, we don't need it. `position_ids` will be put on the same device when we call `.to(device)` on the moduel